kernel/classes/datatypes/ezimage/ezimagetype.php

```diff
@@ -254,6 +258,27 @@ function validateObjectAttributeHTTPInput( $http, $base, $contentObjectAttribute
                 'The size of the uploaded file exceeds the limit set for this site: %1 bytes.' ), $maxSize );
             return eZInputValidator::STATE_INVALID;
         }
+
+        // Check for a valid alternative text if there is an image and
+        // if the alt text is required
+        if(    $contentObjectAttribute->attribute( 'has_content' ) && self::isAltTextRequired( $contentObjectAttribute ) )
+        {
+            $altTextValid = false;
+            if( $http->hasPostVariable( $httpRequiredImageAltTextName ) )
+            {
+                if( self::validateImageAltText( $http->postVariable( $httpRequiredImageAltTextName ) ) )
+                {
+                    $altTextValid = true;
```

Comment:
Is this intermediate step necessary? I don't understand why we can't directly run setValidationError and return STATE_INVALID directly here.